### DISCUSSION TITLE
Two changes that may improve helm-outline

### DISCUSSION THREAD

**Discussion by zbelial** [emacs-helm/helm#2681] [Ideas]
About helm-outline in helm-misc.el, I think there are two changes that can be done to improve it(make it usable in more modes). 
1. Use `outline-regexp` instead of hardcoded "^[*]+".  Some modes ( like markdown-mode, latex-mode..) have a customized value for outline-regexp, and you can even customize it for elisp-mode(below is a picture to show this), go-mode, rust-mode and so on.
2. Delete the `cl-assert` at the beginning of helm-outline. From my experiment, after deleting it and setting `outline-regexp` in a buffer, you can use helm-outlne there, no need to enable outline-minor-mode.

Following is the picture of helm-misc.el where I set outline-regexp to ";;[;*]+[\s\t]+", then helm-outline showed the outline:
![image](https://github.com/user-attachments/assets/bb1704fe-7f2a-47ab-bd9b-c6ca7a30e1c6)

**Comment by thierryvolpiatto** [comment]
太阳 ***@***.***> writes:

> About helm-outline in helm-misc.el, I think there are two changes that can be done to improve it(make it usable in more modes).
>
>  1. Use outline-regexp instead of hardcoded "^[*]+". Some modes ( like markdown-mode, latex-mode..) have a customized value, and you can even define it for elisp-mode(below is a picture to
>     show this)
>  2. Delete the cl-assert at the beginning of helm-outline. From my experiment, after deleting it and setting outline-regexp in a buffer, you can use helm-outlne there, no need to enable
>     outline-minor-mode.

Done, thanks.

-- 
Thierry
